submodule/main.c

```diff
@@ -2,60 +2,174 @@
 #include "Python.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include <immintrin.h>
 
-static long bit_floor_shifted(long x) {
-    long res = 1;
-    while (x > 1) {
-        x >>= 1;
-        res <<= 1;
+#if defined(__OPTIMIZE__)
+
+/* LIKELY(), UNLIKELY() definition */
+/* Checks taken from
```

Comment:
One super tiny nitpick is that you can use up to 100 chars for each line, which might make some of the comments cleaner.

Reply:
That specific comment is pretty nice and clean already but I changed some others.